### PR TITLE
Implemented function returning Wilkinson Matrix

### PR DESCRIPTION
**Pull request by ks147**
#### Brief description of what is fixed or changed
Implemented function returning Wilkinson matrix

#### Release Notes

* matrices
  * Added function returning Wilkinson matrix

### CODE REVIEW

**Comment by ks147** [comment]
@sylee957, please review

File: sympy/matrices/common.py [hunk 1]
```diff
@@ -1208,6 +1208,51 @@ def entry(i, j):
             return kls.zero
         return kls._new(size, size, entry)
 
+    @classmethod
+    def Wilkinson(kls, n):
```
Comment (sylee957) [inline sympy/matrices/common.py:1212]:
```suggestion
    def wilkinson(kls, n):
```
I'd try to use lowercase for names

File: sympy/matrices/common.py [hunk 2]
```diff
@@ -1208,6 +1208,51 @@ def entry(i, j):
...
+        """
+        from sympy.matrices.dense import zeros, diag
+        D = zeros(2*n + 1)
+        for i in range(2*n):
+            D[i, i + 1] = S.One
+
+        wminus = diag([i for i in range(-n, n + 1)], unpack=True) + D + D.T
+        wplus = abs(diag([i for i in range(-n, n + 1)], unpack=True)) + D + D.T
+
+        return wminus, wplus
```
Comment (sylee957) [inline sympy/matrices/common.py:1255]: As I'd noted in #20911, you should consider the class of the output to match the class of the method it's invoked from
So you should check 
```
MutableDenseMatrix.wilkinson -> MutableDenseMatrix
MutableSparseMatrix.wilkinson -> MutableSparseMatrix
ImmutableDenseMatrix.wilkinson -> ImmutableDenseMatrix
ImmutableSparseMatrix.wilkinson -> ImmutableSparseMatrix
```

Comment (ks147) [reply sympy/matrices/common.py]: I've fixed this in latest commit. 
```python
>>> from sympy import MutableSparseMatrix, ImmutableDenseMatrix, ImmutableSparseMatrix, MutableDenseMatrix
>>> w_m, w_p = MutableDenseMatrix.wilkinson(1)
>>> type(w_m)
<class 'sympy.matrices.dense.MutableDenseMatrix'>
>>> w_m, w_p = MutableSparseMatrix.wilkinson(1)
>>> type(w_m)
<class 'sympy.matrices.sparse.MutableSparseMatrix'>
>>> w_m, w_p = ImmutableDenseMatrix.wilkinson(1)
>>> type(w_m)
<class 'sympy.matrices.immutable.ImmutableDenseMatrix'>
>>> w_m, w_p = ImmutableSparseMatrix.wilkinson(1)
>>> type(w_m)
<class 'sympy.matrices.immutable.ImmutableSparseMatrix'>
```

**Pull request merged**